### DISCUSSION TITLE
Error when trying to write global river output file when running with a subgrid (JULES 6.2)

### DISCUSSION THREAD

**Discussion by nicgedney** [MetOffice/simulation-systems#86] [JULES]
I’ve been running a JULES at 6.2 on a sub-grid. I’m aware that the river routing should be run with caution in this case but the land bounds cover the basin I’m interested in. The river routing is still run on the global grid. This works when running straight from a namelist and executable. However it crashes when trying to write a rivers output file after the nml has been converted it to a suite. The rivers output file has been setup as a global grid (as expected) but it looks like the output data may be on?. Does anyone have any ideas what is going on?

The cylc-run output is in ~hadng/cylc-run/test02_vn6.2/

Job.out:
Opening time series with data_period=-1
{MPI Task 0} [INFO] file_ncdf_open: Opening file /data/d05/hadng/jules_output/test02_vn6.2/GSWP3-W5E5_OBSCLIM/isimip3a_pfch4_gswp3-w5e5_obsclim_spinup2_01.rivers.nc for writing
Application 306920146 exit codes: 137
Job.err:
{MPI Task 0} [FATAL ERROR] file_gridded_write_var: At least one dimension of given cube has incorrect size

Code where this error is produced:
vn6.2_trunk/src/io/file_handling/gridded/file_gridded_write_var.inc:

**Comment by richardgilham** [comment]
Hi Nic,

Have you compared to the namelist files generated by the suite when it's run? They're findable in `cylc-run/suite/work/1/jules_app_name`. If you've ported across the content of the files correctly, they ought to be the same...

**Reply by nicgedney** [reply]
There's a bug in 6.2 which is fixed in 6.3.... I think it only appears depending on compile options so am upgrading to 6.3. See Emma's post below

**Reply by richardgilham** [reply]
Any reason you can't upgrade to the latest release? The 6.x series won't be any good for the new Met Office systems.

**Comment by emmrob** [comment]
Hi Nic,

It sounds like you’ve run in to the "dimension check" bug that I introduced in vn6.2 then fixed in vn6.3 with ticket [#1241](https://code.metoffice.gov.uk/trac/jules/ticket/1241). 

As far as I could tell at the time, it only shows up when compiling with debug: _"When this checks the sizes of dimensions, it checks too many dimensions for 1d grids (this seems to only be picked up when compiling with debug rather than normal)"_. Which could explain why you're only seeing it in your suite (if the suite is compiling with debug). But it might also be compiler dependent.

Note that, even if you compile without debug, you might run into the other bug that was fixed by that ticket, where it doesn't close the river output files, so it might fall over because too many files are open at once.

If you're writing river grid output, then I'd recommend upgrading to vn6.3 if at all possible. Otherwise you can fix it by applying the [changes associated with the ticket](https://code.metoffice.gov.uk/trac/jules/changeset?reponame=&new=22018%40main%2Fbranches%2Fdev%2Femmarobinson%2Fvn6.2_file_close&old=21512%40main%2Ftrunk), which are fairly minimal and only affect `src/io/file_handling/gridded/file_gridded_write_var.inc` and `src/io/file_handling/timeseries/file_ts_internal_open_file.inc`.

**Comment by hashmifatima** [comment]
> I’ve been running a JULES at 6.2 on a sub-grid. I’m aware that the river routing should be run with caution in this case but the land bounds cover the basin I’m interested in. The river routing is still run on the global grid. This works when running straight from a namelist and executable. However it crashes when trying to write a rivers output file after the nml has been converted it to a suite. The rivers output file has been setup as a global grid (as expected) but it looks like the output data may be on?. Does anyone have any ideas what is going on?
> 
> The cylc-run output is in ~hadng/cylc-run/test02_vn6.2/
> 
> Job.out: Opening time series with data_period=-1 {MPI Task 0} [INFO] file_ncdf_open: Opening file /data/d05/hadng/jules_output/test02_vn6.2/GSWP3-W5E5_OBSCLIM/isimip3a_pfch4_gswp3-w5e5_obsclim_spinup2_01.rivers.nc for writing Application 306920146 exit codes: 137 Job.err: {MPI Task 0} [FATAL ERROR] file_gridded_write_var: At least one dimension of given cube has incorrect size
> 
> Code where this error is produced: vn6.2_trunk/src/io/file_handling/gridded/file_gridded_write_var.inc:

Hi
I am running jules vn7.4. I want to create river ancil file. how to create river ancil for regional domain?
